test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs

```diff
@@ -760,14 +684,149 @@ private static void CounterUpdateThread<T>(object obj)
             }
         }
 
+        private static void HistogramUpdateThread<T>(object obj)
+            where T : struct, IComparable
+        {
+            if (obj is not UpdateThreadArguments<T> arguments)
+            {
+                throw new Exception("Invalid args");
+            }
+
+            var mre = arguments.MreToBlockUpdateThread;
+            var mreToEnsureAllThreadsStart = arguments.MreToEnsureAllThreadsStart;
+            var histogram = arguments.Instrument as Histogram<T>;
+
+            if (Interlocked.Increment(ref arguments.ThreadsStartedCount) == numberOfThreads)
+            {
+                mreToEnsureAllThreadsStart.Set();
+            }
+
+            // Wait until signalled to start calling update on aggregator
+            mre.WaitOne();
+
+            for (int i = 0; i < numberOfMetricUpdateByEachThread; i++)
+            {
+                for (int j = 0; j < arguments.ValuesToRecord.Length; j++)
+                {
+                    histogram.Record(arguments.ValuesToRecord[j]);
+                }
+            }
+        }
+
+        private void MultithreadedCounterTest<T>(T deltaValueUpdatedByEachCall)
+            where T : struct, IComparable
+        {
+            var metricItems = new List<Metric>();
+            var metricReader = new BaseExportingMetricReader(new InMemoryExporter<Metric>(metricItems));
+
+            using var meter = new Meter($"{Utils.GetCurrentMethodName()}.{typeof(T).Name}.{deltaValueUpdatedByEachCall}");
+            using var meterProvider = Sdk.CreateMeterProviderBuilder()
+                .AddMeter(meter.Name)
+                .AddReader(metricReader)
+                .Build();
+
+            var argToThread = new UpdateThreadArguments<T>
+            {
+                ValuesToRecord = new T[] { deltaValueUpdatedByEachCall },
+                Instrument = meter.CreateCounter<T>("counter"),
+                MreToBlockUpdateThread = new ManualResetEvent(false),
+                MreToEnsureAllThreadsStart = new ManualResetEvent(false),
+            };
+
+            Thread[] t = new Thread[numberOfThreads];
+            for (int i = 0; i < numberOfThreads; i++)
+            {
+                t[i] = new Thread(CounterUpdateThread<T>);
+                t[i].Start(argToThread);
+            }
+
+            argToThread.MreToEnsureAllThreadsStart.WaitOne();
```

Comment:
Would we need try/catch here to handle potential exceptions being thrown?

Reply:
I don't think so, if an exception is thrown the test will be considered failed